api/collection.go

```diff
@@ -78,6 +79,23 @@ func (api *API) AddCollectionHandler(w http.ResponseWriter, r *http.Request) {
 	log.Event(ctx, "add collection request completed successfully", log.INFO, logData)
 }
 
+func (api *API) validateCollection(ctx context.Context, collection *models.Collection) error {
+
+	if len(collection.Name) == 0 {
+		return collections.ErrCollectionNameEmpty
+	}
+
+	_, err := api.collectionStore.GetCollectionByName(ctx, collection.Name)
```

Comment:
Question: Does this need to take into consideration old collections or collections that have already been published? I think its fairly common for the publishing team to reuse names for collections - like Weekly Deaths or something that is output on a regular basis.

I think this check may be too restrictive in that you can only ever have a collection called `X` once. Does this check need look for an `active` collection with name X?

I appreciate I am not involved in any of the planning or discussions you've had around this so sorry if this is a silly question but I felt I should ask in case this was an oversight.

Reply:
Quite possibly - I asked this at the start of the task as I imagined that collection names would be reused. The question is being fed back to the product team to see if it's something we want to keep or not. We may have to change this in future like you say, to only check active collections or something. For now we have gone with the simplest option while it's considered.